### ISSUE TITLE
Y24-186 - [BUG] GBS Stock Manifest upload created duplicate samples

### ISSUE THREAD

**Issue by andrewsparkes** [sanger/sequencescape#4193]
**Describe the bug**
A GBS Stock plate manifest containing 41 plates was uploaded and samples created then 6 minutes later a duplicated set of samples were created in the wells. See events. 2 aliquots each with the same sample per well. Manifest id 24721.

**RT Ticket Number**
806313

**To Reproduce**
Steps to reproduce the behaviour:
Neil tried to replicate this in Training but was unable to do so.
Suggests an inconsistent error.

**Expected behaviour**
The re-upload (if that is indeed what they did, check logs) should not have allowed a second set of aliquots and samples to be created in the existing wells.

**Additional context**
Neil will correct the data for this manifest in production.
These GBS Stock plates were the only ones with duplicated samples. i.e. the manifest normally works.
So this might be difficult to replicate and fix.

**Comment by BenTopping** [comment]
Research doc: https://docs.google.com/document/d/1lqEPpsqYH2tYW3xytTirgTZrCWbAbRTrXYag5o7uT5g/edit?usp=sharing

TLDR: Suspected edge case concurrency issue when uploading large manifests consecutively.

**Comment by KatyTaylor** [comment]
Good investigation, thanks! Does seem fairly low priority to fix it, seeing as it has only happened once that we're aware of. We could warn the SSRs not to re-upload until the first upload has finished? Presumably the UI shows you when it's finished?

**Comment by BenTopping** [comment]
> Good investigation, thanks! Does seem fairly low priority to fix it, seeing as it has only happened once that we're aware of. We could warn the SSRs not to re-upload until the first upload has finished? Presumably the UI shows you when it's finished?

Yes I don't think it is urgent, maybe I should make a story in case it comes up again in the future? If it does happen again it is also fairly trivial to manually fix.
The UI just waits until it is done then redirects the user to the manifests page with an alert on whether it was successful or not.

**Comment by andrewsparkes** [comment]
Added a few comments.
Good investigation and impressed you managed to replicate the issue.
Bit surprised it is so slow to process the files. Think the user just thought it was hanging or had failed and re-uploaded it.

**Comment by BenTopping** [comment]
Closing in favour of the follow on bug fix for this story https://github.com/sanger/sequencescape/issues/4266.
